lib/widgets/HomeScreen/home_screen.dart

```diff
@@ -0,0 +1,87 @@
+import 'package:flutter/material.dart';
+import 'package:classroom_app/styles/color/colors.dart';
+import 'package:classroom_app/widgets/CustomButton/custom_button.dart';
+
+class HomeScreen extends StatelessWidget {
+  const HomeScreen({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    double screenWidth = MediaQuery.of(context).size.width;
+    double screenHeight = MediaQuery.of(context).size.height;
```

Comment:
MediaQuery.of(context).size is called multiple times, which can slightly impact performance.
```suggestion
    final size = MediaQuery.of(context).size;
    final screenWidth = size.width;
    final screenHeight = size.height;
```

Reply:
1. Performance Impact:

- MediaQuery.of(context) is relatively fast because it doesn't involve complex computation. It simply retrieves the MediaQueryData from the context, which contains information about the screen's size, device pixel ratio, text scaling factor, and other properties.

- Accessing properties like .size.width or .size.height from the MediaQueryData object is a direct operation and is also not computationally expensive.

Thus, calling MediaQuery.of(context) multiple times in a single widget build method will not have any noticeable performance issues in typical applications. Flutter is optimized for these types of calls, and they're not expensive in terms of CPU or memory usage.

But I am changing for best practice

Reply:
ditto

Reply:
done